### ISSUE TITLE
Bug: ClassStaticBlockScope is not exported from @typescript-eslint/scope-manager

### ISSUE THREAD

**Issue by zaewc** [typescript-eslint/typescript-eslint#12459]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

scope-manager

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.3&fileType=.tsx&code=JYWwDg9gTgLgBAbzgYQDYEMDOmDKN0zADGAQqhEQNY5ERgCmcAvnAGZQQhwDkAAjAE8GmIlGBgYAWnqZUwAHYwA9CLr1JIdPPQBzelG4BuIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
import { ClassStaticBlockScope } from '@typescript-eslint/scope-manager';


// Module '"@typescript-eslint/scope-manager"' has no exported member 'ClassStaticBlockScope'.
```

### ESLint Config

not a lint-rule issue; pure export omission

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

`ClassStaticBlockScope` should be exported from the package, like every other scope class. It's part of the public `Scope` union([packages/scope-manager/src/scope/Scope.ts](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/scope-manager/src/scope/Scope.ts)) and `ScopeManager` instantiates it via `nestClassStaticBlockScope`, so it's a real member of the public API surface. All 18 other scope classes (`BlockScope`, `ClassScope`, `ConditionalTypeScope` …) are re-exported from [packages/scope-manager/src/scope/index.ts](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/scope-manager/src/scope/index.ts).

### Actual Result

`ClassStaticBlockScope` cannot be imported from `@typescript-eslint/scope-manager`

TypeScript reports:

```ts
Module '"@typescript-eslint/scope-manager"' has no exported member 'ClassStaticBlockScope'.
```
Looking at [packages/scope-manager/src/scope/index.ts](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/scope-manager/src/scope/index.ts), the barrel exports `ClassScope` and then `ConditionalTypeScope`, but does not export `ClassStaticBlockScope`:
```ts
export * from './ClassScope';
// missing
export * from './ConditionalTypeScope';
```

`git log -S 'ClassStaticBlockScope' -- packages/scope-manager/src/scope/index.ts` returns no results, so it looks like this was missed when class static block support was added rather than intentionally omitted

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/scope-manager`     | `8.62.1` |
| `TypeScript`                           | `6.0.3` |
| `ESLint`                               | `10.5.0` |
| `node`                                 | `20.20.2` |

**Comment by zaewc** [comment]
I'd like to work on this and send a PR
